### PR TITLE
[E2E] Add simple rest service for e2e tests

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary

This PR introduces a simple REST service to be used in end-to-end (E2E) tests as a replacement for the LLM service.

It utilizes the [davarski/go-rest-api-demo](https://github.com/adavarski/REST-go-k8s-example) Docker image to provide this service.

- [ ] Add a follow-up PR to enable this service on CI

## Issue

Due to E2E test flakiness, mostly observed in CI, which is likely caused by the resource-intensive LLM service, we decided to use a simple REST service for testing this feature.

- #723

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

## Summary by CodeRabbit

- **New Features**
	- Introduced a new REST service configuration, enhancing the app's capability to interact through RESTful APIs.
	- Added service and endpoint definitions for REST, supporting broader functionality in communication.

- **Bug Fixes**
	- Improved request handling logic in the testing suite to dynamically set HTTP methods based on input.

- **Tests**
	- Updated test scenarios to reflect changes in service type and request paths, ensuring accurate testing parameters.
	- Added convenience method in the test suite for easier testing of GET requests without data.

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -163,6 +167,14 @@ genesis:
...
                 id: anvil
                 name: ""
+            - endpoints:
+                - configs: []
+                  rpc_type: REST
+                  url: http://relayminer1:8545
+              service:
+                compute_units_per_relay: 1
+                id: rest
+                name: ""
```
Comment (Olshansk) [inline config.yml:177]: Just the id is fine

Comment (Olshansk) [reply config.yml:177]: Ditto above

File: e2e/tests/node.go [hunk 2]
```diff
@@ -176,9 +176,13 @@ func (p *pocketdBin) runCurlPostCmd(rpcUrl, service, path, data string, args ...
...
-		"-X", "POST", // HTTP method
+		"-X", method, // HTTP method
 		"-H", "Content-Type: application/json", // HTTP headers
-		"--data", data, urlStr, // POST data
+		urlStr,
+	}
+
+	if method == "POST" {
+		base = append(base, "--data", data)
 	}
```
Comment (Olshansk) [inline e2e/tests/node.go:186]:
```suggestion
	} else if len(data) != 0 {
		fmt.println(fmt.Sprtinf("WARN: data provided but not being included in the %s request", method))
	}
```

File: e2e/tests/init_test.go [hunk 3]
```diff
@@ -418,7 +418,13 @@ func (s *suite) TheApplicationSendsTheSupplierASuccessfulRequestForServiceWithPa
 	// that allows any application to send a relay in LocalNet and our E2E Tests.
 	require.Equal(s, "app1", appName, "TODO_HACK: The LocalNet AppGateServer is self_signing and only supports app1.")
 
-	res, err := s.pocketd.RunCurlWithRetry(appGateServerUrl, serviceId, path, requestData, 5)
+	method := "POST"
+	// If requestData is empty, assume a GET request
+	if requestData == "" {
+		method = "GET"
+	}
```
Comment (Olshansk) [inline e2e/tests/init_test.go:425]: Same as below, can you add a println statement in case data is provided but we're ignoring it?

**Review by Olshansk — APPROVED** [review]
Preliminary approval with a few minor tweaks.

File: e2e/tests/init_test.go (hunk 3 above)
Comment (red-0ne) [reply e2e/tests/init_test.go:425]: I don't this it applies to this case. We are using `requestData` to infer the `method` here.

**Comment by Olshansk** [comment]
@red-0ne Could we pick this one up?

**Pull request merged**